{
  if (context == null) {
    return;
  }
  try {
    context.close();
  }
 catch (  Throwable t) {
    log.info("Ignoring a problem which occurred while closing: " + context,t);
  }
  context=null;
}
